model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java

```diff
@@ -164,10 +205,106 @@ public LanguageDetection getWholeTextLanguageDetection() {
+                        cmListBuilder.extendedProperty(key.substring(8), me.getValue());
+                    }
+                }
+            }
+
+            compatMentions = cmListBuilder.build();
+        }
+        return compatMentions;
+    }
+
+    private void downconvertEntities(List<EntityMention> entityMentionList, ListAttribute<Entity> entities) {
+        for (Entity entity : entities) {
+            if (entity.getMentions() != null) {
+                for (Mention mention : entity.getMentions()) {
+                    // If the conversion process stashed a per-mention type, recover it here.
+                    String type = (String) mention.getExtendedProperties().get("old-entity-type");
+                    if (type == null) {
+                        // In the new model, it's on the Entity.
+                        type = entity.getType();
+                    }
+
+                    EntityMention.Builder emBuilder = new EntityMention.Builder(mention.getStartOffset(),
+                            mention.getEndOffset(),
+                            type);
+
+                    if (mention.getConfidence() != null) {
+                        emBuilder.confidence(mention.getConfidence());
+                    }
+                    if (mention.getExtendedProperties() != null && mention.getExtendedProperties().size() > 0) {
+                        for (Map.Entry<String, Object> me : mention.getExtendedProperties().entrySet()) {
+                            if (me.getKey().equals("old-entity-type")) {
+                                //
+                            } else if (me.getKey().equals("oldFlags")) {
+                                emBuilder.flags((Integer) me.getValue());
+                            } else if (me.getKey().equals("oldCoreferenceChainId")) {
+                                emBuilder.coreferenceChainId((Integer) me.getValue());
+                            } else {
+                                emBuilder.extendedProperty(me.getKey(), me.getValue());
+                            }
+                        }
+                    }
+                    if (mention.getNormalized() != null) {
+                        emBuilder.normalized(mention.getNormalized());
+                    }
+                    if (mention.getSource() != null) {
+                        emBuilder.source(mention.getSource());
+                    }
+                    if (mention.getSubsource() != null) {
+                        emBuilder.subsource(mention.getSubsource());
+                    }
+
+                    entityMentionList.add(emBuilder.build());
+                }
+            }
+        }
+
+        // return these mentions in document order.
+        Collections.sort(entityMentionList, new Comparator<EntityMention>() {
+            @Override
+            public int compare(EntityMention o1, EntityMention o2) {
+                return o1.getStartOffset() - o2.getStartOffset();
```

Comment:
Add endOffset too, to define an order for mentions starting in same place?
I guess we could then add mentionText if they overlap exactly.
It may happen?  (Does redactor act on ADM or beneath?)